File: bubbleSort.cpp
#include <iostream>
#include <vector>

int main(){
    std::vector<int> toSort = {3, 9, 7, 10, 2, 0, 6, 1};
    bool isSorted = false;
    
    while (!isSorted){
        isSorted = true;
        for (int i = 0; i < toSort.size() - 1; i++){
            if (toSort[i] > toSort[i + 1]){
                std::swap(toSort[i], toSort[i +1]);
                isSorted = false;
            }
        }
            
    }
    for (int i = 0; i < toSort.size(); i++){
        std::cout << toSort[i] << "\n";
    }

    return 0;
}

//Time: O(n^2)
//Space: O(1)

File: bubbleSort.py
def bubbleSort():
    toSort = [6, 3, 10, 2, 9, 7, 1]
    isSorted = False
    
    print("unsorted: ")
    for i in range(0, len(toSort)):
        print(toSort[i])
    print("\n")
    
    while (not isSorted):
        isSorted = True
        for i in range(0, len(toSort) - 1):
            if toSort[i] > toSort[i + 1]:
                #swap(toSort[i], toSort[i + 1])
                temp = toSort[i]
                toSort[i] = toSort[i + 1]
                toSort[i + 1] = temp
                
                isSorted = False
    
    
    print("isSorted")
    for i in range(0, len(toSort)):
        print(toSort[i])
    print("\n")

    
bubbleSort()


#Time: O(n^2)
#Space: O(n)
